src/find_controls.cpp

```diff
@@ -225,26 +229,34 @@ FindMode::Values GetFindModeValueImpl2()
     const auto val = static_cast<FindMode::Values>(stoul(str));
     if(FindMode::Values::Begin_ <= val && val < FindMode::Values::End_)
         return val;
-    MIOPEN_LOG_NQE("Wrong MIOPEN_FIND_MODE, using default.");
-    return FindMode::Values::Default_;
+    MIOPEN_LOG_NQE("Wrong " << variable.GetName() << ", using default.");
+    return std::nullopt;
 }
 
-FindMode::Values GetFindModeValueImpl()
+template <class Variable>
+FindMode::Values GetFindModeValue(Variable variable, FindMode::Values defaultValue)
 {
-    auto rv = GetFindModeValueImpl2();
-    MIOPEN_LOG_NQI("MIOPEN_FIND_MODE = " << rv);
-    return rv;
-}
-
-FindMode::Values GetFindModeValue()
-{
-    static const FindMode::Values val = GetFindModeValueImpl();
+    static const FindMode::Values val = [&]() {
+        auto rv = GetFindModeValueImpl2(variable).value_or(defaultValue);
+        MIOPEN_LOG_NQI(variable.GetName() << " = " << rv);
+        return rv;
+    }();
```

Comment:
It seems we have to forbid any runtime environmental values updates.
Looks like this code caches env variable state one more time and thus makes any runtime update impossible.

I'm not against this particular change, I'd like to say that we have to focus on our tests and testing infrastructure and properly fix it asap. (though, there are some possible workarounds for this env var problem even for a single-binary test)

@junliume @JehandadKhan

Reply:
Yes we will need to log this as part of the test infra renovations.